platform/posix/transport/src/mbedtls_pkcs11_posix.c

```diff
@@ -0,0 +1,958 @@
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = MBEDTLS_PKCS11_INTERNAL_ERROR;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsPkcs11Status_t configureMbedtlsFragmentLength( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context )
+{
+    MbedtlsPkcs11Status_t returnStatus = MBEDTLS_PKCS11_SUCCESS;
+
+    /* Set Maximum Fragment Length if enabled. */
+    #ifdef MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
+        int32_t mbedtlsError = 0;
+
+        /* Enable the max fragment extension. 4096 bytes is currently the largest fragment size permitted.
+         * See RFC 8449 https://tools.ietf.org/html/rfc8449 for more information.
+         *
+         * Smaller values can be found in "mbedtls/include/ssl.h".
+         */
+        mbedtlsError = mbedtls_ssl_conf_max_frag_len( &( pMbedtlsPkcs11Context->config ), MBEDTLS_SSL_MAX_FRAG_LEN_4096 );
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to maximum fragment length extension: mbedTLSError= %s : %s.",
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = MBEDTLS_PKCS11_INTERNAL_ERROR;
+        }
+    #endif /* ifdef MBEDTLS_SSL_MAX_FRAGMENT_LENGTH */
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t generateRandomBytes( void * pCtx,
+                                    unsigned char * pRandom,
+                                    size_t randomLength )
+{
+    /* Must cast from void pointer to conform to MbedTLS API. */
+    MbedtlsPkcs11Context_t * pContext = ( MbedtlsPkcs11Context_t * ) pCtx;
+    CK_RV xResult;
+
+
+    xResult = pContext->pP11FunctionList->C_GenerateRandom( pContext->p11Session, pRandom, randomLength );
+
+    if( xResult != CKR_OK )
+    {
+        LogError( ( "Failed to generate random bytes from the PKCS #11 module." ) );
+    }
+
+    return ( int32_t ) xResult;
+}
+
+/*----------------------------------------------------------*/
+
+static int32_t readCertificateIntoContext( MbedtlsPkcs11Context_t * pContext,
+                                           char * pLabelName,
+                                           CK_OBJECT_CLASS class,
+                                           mbedtls_x509_crt * pCertificateContext )
+{
+    CK_RV pkcs11Ret = CKR_OK;
+    CK_ATTRIBUTE template = { 0 };
+    CK_OBJECT_HANDLE certificateHandle = 0;
+    int32_t mbedtlsRet = 0;
+
+    /* Get the handle of the certificate. */
+    pkcs11Ret = xFindObjectWithLabelAndClass( pContext->p11Session,
+                                              pLabelName,
+                                              strlen( pLabelName ),
+                                              class,
```

Comment:
Seems like this can just be `CKO_CERTIFICATE` instead of being passed as an input parameter?